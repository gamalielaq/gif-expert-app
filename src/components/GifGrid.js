import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from "../hooks/useFetchGifs"
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const { data, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className='animate__animated animate__fadeIn'>{category}</h3>

            { loading && 'Cargando....'}

            <div className='card-grid'>
                {
                    data.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.protoType = {
    category: PropTypes.string.isRequired
}