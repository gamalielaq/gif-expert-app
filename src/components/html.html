<!DOCTYPE html>
<html lang="es">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    <div class="container"> 
        <h1>Hellow world</h1>
        <div class="main">
            Esto es el contenido de mi Web
        </div>
    </div>
    <footer>
        <div class="footer">
                <h5>Gamaliel Humberto Abanto Quiroz</h5>
        </div>
    </footer>
</body>
</html>