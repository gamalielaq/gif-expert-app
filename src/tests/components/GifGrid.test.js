import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
import React from 'react';
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en GifGrid Component', () => {

    const category = 'One Punch';

    test('Deve mostrar el compoente corretamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: false
        })

        const wrapper = shallow(< GifGrid category= {category} />);
        expect( wrapper ).toMatchSnapshot();
    })


    test('Deve de mostrar items cuando se cargan imÃ¡genes', () => {

        const gifs = [{
            id: 'abc',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(< GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();

    })

});