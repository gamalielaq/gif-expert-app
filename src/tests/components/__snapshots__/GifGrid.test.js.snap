// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Component Deve de mostrar items cuando se cargan im√°genes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en GifGrid Component Deve mostrar el compoente corretamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
